core/Pos/Binary/Class/Primitive.hs

```diff
@@ -123,13 +107,52 @@ deserializeOrFail bs0 =
         BSL.Empty           -> k Nothing      >>= supplyAllInput BSL.Empty
     supplyAllInput _ (CBOR.Read.Fail bs _ exn) = return (Left (exn, bs))
 
--- | Strict variant of 'deserializeOrFail'.
-deserializeOrFail'
-    :: Bi a
-    => BS.ByteString
-    -> Either (CBOR.Read.DeserialiseFailure, BS.ByteString)
-              (a, BS.ByteString)
-deserializeOrFail' = deserializeOrFail . BSL.fromStrict
+    -- | Deserialize a Haskell value from the external binary representation.
+    --
+    -- This allows /input/ data to be provided incrementally, rather than all in one
+    -- go. It also gives an explicit representation of deserialisation errors.
+    --
+    -- Note that the incremental behaviour is only for the input data, not the
+    -- output value: the final deserialized value is constructed and returned as a
+    -- whole, not incrementally.
+    deserializeIncremental :: Bi a => ST s (CBOR.Read.IDecode s a)
+    deserializeIncremental =
+        CBOR.Read.deserialiseIncremental (runReaderT decode dc)
+
+-- | Try to deserialize value or fail.
+--
+-- /Throws/: @'CBOR.Read.DeserialiseFailure'@ if the given external
+-- representation is invalid or does not correspond to a value of the
+-- expected type.
+deserializeThrow :: (Bi a) => LByteString -> a
+deserializeThrow =
+    either throw identity . bimap fst fst . deserializeOrFailRaw def
+
+decodeFullProcess ::
```

Comment:
I want a haddock for this

Reply:
I'll add a regular comment since this function is not exported.